packages/slate/src/interfaces/editor/isBlock.ts

```diff
@@ -2,10 +2,11 @@ import { Editor } from 'slate';
 
 import type { TEditor, Value } from './TEditor';
 
-import { isElement } from '../element';
+import { type TElement, isElement } from '../element';
 
 /** Check if a value is a block `Element` object. */
 export const isBlock = <V extends Value>(
   editor: TEditor<V>,
   value: any
-): boolean => isElement(value) && Editor.isBlock(editor as any, value);
+): value is TElement =>
```

Comment:
This may casue issues since the type guard doesn't precisely match the actual return value. If `isBlock` is used in a conditional, inside the negative branch TypeScript will believe that the value definitely isn't a `TElement`.

Here's a [TS Playground](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAKgogGwgWwgO2FAvFA3lUSALigGdgAnASzQHMoBfAKEOhhggA9Md9gvgJctTqMW4NgDkA9gBNoOeElQYoAH1gduTJgGNpaclABGCaboDWMCaWxQA2gHJT5i44C6eg0aqkAQmaWdgAUAG4AhggArhAk4WggAJQkEdHQvrCIKOg8AHxMUFCOrI5QNFCpMVAAZNUmgVY2AHQ08pwA8gBmYZExTayJUACEWDgAtACMOvqGmLQQwDLyIcmwSwq5UMF8EiTODaUMidPemMgg63bzi3IQwcdUnVu+Aa7B5+uJg7gFUDNG51eQRwH1uv0BDSYDCgEAQpGgP0KAHoAFQo35QFGwCQAZV01DAmGMsKoEFCEFswAAFuFMNToKwoNInqC0ECLGVbPJOjQqPwECBNAIADQmKLADFY8oUCCRPmCvlQZDhQXEqDxMpoBA0aCw7IYJqSpG-f5nC4Gdl2UHycHmtmQhhAA) demonstrating the problem.

Reply:
That is correct, will stick to using `isBlock() && isElement()`